### ISSUE TITLE
expo:app Will not run `run-ios` on fresh install.

### ISSUE THREAD

**Issue by TheSavageDev** [nrwl/nx#14873]
### Current Behavior

On a  fresh install of a @nrwl/expo application, `nx run-ios app` and `nx run-android app` do not work.

### Expected Behavior

The app should run on an fresh install.

### GitHub Repo

https://github.com/TheSavageDev/nx-expo-fail

### Steps to Reproduce

1. Create a new integrated workspace
2. Install @nrwl/expo `yarn add -D @nrwl/expo`
3. Create an expo app `nx g @nrwl/expo:app app`
4. Run `nx run-ios app`

### Nx Report

```shell
>  NX   Report complete - copy this into the issue template

   Node : 16.14.2
   OS   : darwin arm64
   yarn : 1.22.19
   
   nx : 15.6.3
   @nrwl/angular : Not Found
   @nrwl/cypress : Not Found
   @nrwl/detox : 15.6.3
   @nrwl/devkit : 15.6.3
   @nrwl/esbuild : Not Found
   @nrwl/eslint-plugin-nx : 15.6.3
   @nrwl/expo : 15.6.3
   @nrwl/express : Not Found
   @nrwl/jest : 15.6.3
   @nrwl/js : 15.6.3
   @nrwl/linter : 15.6.3
   @nrwl/nest : Not Found
   @nrwl/next : Not Found
   @nrwl/node : Not Found
   @nrwl/nx-cloud : 15.0.3
   @nrwl/nx-plugin : Not Found
   @nrwl/react : 15.6.3
   @nrwl/react-native : Not Found
   @nrwl/rollup : Not Found
   @nrwl/schematics : Not Found
   @nrwl/storybook : Not Found
   @nrwl/web : Not Found
   @nrwl/webpack : 15.6.3
   @nrwl/workspace : 15.6.3
   @nrwl/vite : Not Found
   typescript : 4.8.4
   ---------------------------------------
   Local workspace plugins:
   ---------------------------------------
   Community plugins:
```

### Failure Logs

```shell
First run returns:


> nx run app:run-ios

info Added entries to 'package.json' for 'app' (for autolink):
  "@nrwl/expo": "*"
  "@expo/metro-config": "*"
  "react": "*"
  "@expo/webpack-config": "*"

📝  iOS Bundle Identifier Learn more

✔ What would you like your iOS bundle identifier to be? … com.savagedev.app

✔ Created native project | gitignore skipped
› Metro skipped: Project metro.config.js does not match prebuild template.
› Ensure the project uses @expo/metro-config.
  Learn more
✔ Updated package.json and added index.js entry point for iOS and Android
› Installing using npm
Error: npm exited with non-zero code: 1
Error: npm exited with non-zero code: 1
    at ChildProcess.completionListener (/Users/jasonsavage/segistics/node_modules/@expo/cli/node_modules/@expo/package-manager/node_modules/@expo/spawn-async/build/spawnAsync.js:41:23)
    at Object.onceWrapper (node:events:646:26)
    at ChildProcess.emit (node:events:526:28)
    at maybeClose (node:internal/child_process:1092:16)
    at Process.ChildProcess._handle.onexit (node:internal/child_process:302:5)
    ...
    at spawnAsync (/Users/jasonsavage/segistics/node_modules/@expo/cli/node_modules/@expo/package-manager/node_modules/@expo/spawn-async/build/spawnAsync.js:7:23)
    at NpmPackageManager._runAsync (/Users/jasonsavage/segistics/node_modules/@expo/cli/node_modules/@expo/package-manager/build/NodePackageManagers.js:133:51)
    at NpmPackageManager.installAsync (/Users/jasonsavage/segistics/node_modules/@expo/cli/node_modules/@expo/package-manager/build/NodePackageManagers.js:59:20)
    at NpmPackageManager.addWithParametersAsync (/Users/jasonsavage/segistics/node_modules/@expo/cli/node_modules/@expo/package-manager/build/NodePackageManagers.js:68:25)
    at installPackagesAsync (/Users/jasonsavage/segistics/node_modules/@expo/cli/build/src/install/installAsync.js:82:26)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Object.prebuildAsync (/Users/jasonsavage/segistics/node_modules/@expo/cli/build/src/prebuild/prebuildAsync.js:75:9)
    at async Object.ensureNativeProjectAsync (/Users/jasonsavage/segistics/node_modules/@expo/cli/build/src/run/ensureNativeProject.js:24:9)
    at async runIosAsync (/Users/jasonsavage/segistics/node_modules/@expo/cli/build/src/run/ios/runIosAsync.js:47:9)

 >  NX   1



 ———————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 >  NX   Ran target run-ios for project app (6s)
 
    ✖    1/1 failed
    ✔    0/1 succeeded [0 read from cache]
 
   View structured, searchable error logs at https://nx.app/runs/K0WbsE1lKE
```

Second run compiles some pods and returns:

```
❌  (/Users/jasonsavage/segistics/apps/app/ios/App/AppDelegate.mm:36:3)

  34 | - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
  35 | {
> 36 |   RCTAppSetupPrepareApp(application);
     |   ^ no matching function for call to 'RCTAppSetupPrepareApp'
  37 | 
  38 |   RCTBridge *bridge = [self.reactDelegate createBridgeWithDelegate:self launchOptions:launchOptions];
  39 | 

    Run script build phase '[CP-User] Generate app.config for prebuilt Constants.manifest' will be run during every build because it does not specify any outputs. To address this warning, either add output dependencies to the script phase, or configure it to run in every build by unchecking "Based on dependency analysis" in the script phase. (in target 'EXConstants' from project 'Pods')

› 1 error(s), and 3 warning(s)

CommandError: Failed to build iOS project. "xcodebuild" exited with error code 65.

 >  NX   1



 ———————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 >  NX   Ran target run-ios for project app (1m)
 
    ✖    1/1 failed
    ✔    0/1 succeeded [0 read from cache]
 
   View structured, searchable error logs at https://nx.app/runs/XSIoSVL6g8
```
```


### Additional Information

_No response_

**Comment by Freire71** [comment]
I'm facing the same issue but I'm not using Expo. It does not make sense to have a crash upon a fresh install

**Comment by TheSavageDev** [comment]
Has there been any insight into this? I may have to drop Nx in favor of another tool if not.

**Comment by xiongemi** [comment]
fixed in pr https://github.com/nrwl/nx/pull/15582

you could also run ios app locally by just running `nx start <your app> --ios`.

**Comment by floorz** [comment]
> fixed in pr #15582
> 
> you could also run ios app locally by just running `nx start <your app> --ios`.

This is merged to master but not in any recent tags? How would I get this update?
